File: Fb_ph/Practice/Python/Intersection of Two Arrays.py
'''
Given two integer arrays nums1 and nums2, return an array of their intersection. \
Each element in the result must be unique and you may return the result in any order.

Example 1:

Input: nums1 = [1,2,2,1], nums2 = [2,2]
Output: [2]
'''
'''
TC:- O(n)
SC:- O(n)
'''
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        set1 = set(nums1)
        set2 = set(nums2)
        Result = []
        for num in set2:
            if num in set1 and num in set2:
                Result.append(num)
        return Result

File: C++/Set/Intersection-Of-Two-Arrays.cpp
#include <iostream>
#include <set>
#include <vector>
using namespace std;

class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) 
    {
        set<int>check_intersection;
        vector<int>result;
        for(int i=0;i<nums1.size();i++)
        {
            check_intersection.emplace(nums1[i]);
        }
        for(int i=0;i<nums2.size();i++)
        {
            if (check_intersection.find(nums2[i]) != check_intersection.end())
            {
                if ( !(find(result.begin(),result.end(),nums2[i]) != result.end()))
                    result.emplace_back(nums2[i]);
            }
        }
        return result;
        
    }
};
/*99.62 % faster */
